NXP Boards that do not build hello_world

**Describe the bug**
A small section of boards from boards/nxp do not seem to build hello_world.
This was tested on the latest commit of main1fe904539d32b5b7d18361d63acc7a47c9171400

imx8mn_evk/mimx8mn6/m7
imx8mq_mimx8mq6/a53
imx8qm_mek/mimx8qm6/a72
imx8qm_mek/mimx8qm6/a53
imx8qm_mek/mimx8qm6/m4
imx8qxp_mek/mimx8qx6/a35
imx8qxp_mek/mimx8qx6/m4
imx8ulp_evk/mimx8ud7/f1_dsp
imx8ulp_evk/mimx8ud7/a35

**To Reproduce**
west build -p always -b <board_name> /path/to/samples/hello_world

**Expected behavior**
A successful build of hello world for <board_name>

**Impact**
Delay of CI confirmation and platform testing when PR for Issue [https://github.com/zephyrproject-rtos/zephyr/issues/78353] is posted.

**Logs and console output**
My Environment
- |
   |- zephyr/*
   | 
   |- build/

Example output:  /> west build -p always -b imx8mn_evk/mimx8mn6/m7 ./zephyr/samples/hello_world
-- west build: making build dir /home/emilio/zephyrproject/build pristine
-- west build: generating a build system
Loading Zephyr default modules (Zephyr base).
-- Application: /home/emilio/zephyrproject/zephyr/samples/hello_world
-- CMake version: 3.29.2
-- Found Python3: /home/emilio/zephyrproject/.venv/bin/python3 (found suitable version "3.10.12", minimum required is "3.10") found components: Interpreter
-- Cache files will be written to: /home/emilio/.cache/zephyr
-- Zephyr version: 3.7.99 (/home/emilio/zephyrproject/zephyr)
-- Found west (found suitable version "1.2.0", minimum required is "0.14.0")
-- Board: imx8mn_evk, qualifiers: mimx8mn6/m7
-- ZEPHYR_TOOLCHAIN_VARIANT not set, trying to locate Zephyr SDK
-- Found host-tools: zephyr 0.16.5 (/home/emilio/zephyr-sdk-0.16.5)
-- Found toolchain: zephyr 0.16.5 (/home/emilio/zephyr-sdk-0.16.5)
-- Found Dtc: /home/emilio/zephyr-sdk-0.16.5/sysroots/x86_64-pokysdk-linux/usr/bin/dtc (found suitable version "1.6.0", minimum required is "1.4.6")
-- Found BOARD.dts: /home/emilio/zephyrproject/zephyr/boards/common/stub.dts
-- Generated zephyr.dts: /home/emilio/zephyrproject/build/zephyr/zephyr.dts
-- Generated pickled edt: /home/emilio/zephyrproject/build/zephyr/edt.pickle
-- Generated zephyr.dts: /home/emilio/zephyrproject/build/zephyr/zephyr.dts
-- Generated devicetree_generated.h: /home/emilio/zephyrproject/build/zephyr/include/generated/zephyr/devicetree_generated.h
-- Including generated dts.cmake file: /home/emilio/zephyrproject/build/zephyr/dts.cmake
Parsing /home/emilio/zephyrproject/zephyr/Kconfig
Loaded configuration '/home/emilio/zephyrproject/zephyr/samples/hello_world/prj.conf'
Configuration saved to '/home/emilio/zephyrproject/build/zephyr/.config'
Kconfig header saved to '/home/emilio/zephyrproject/build/zephyr/include/generated/zephyr/autoconf.h'
CMake Error at /home/emilio/zephyrproject/zephyr/cmake/modules/arch_v2.cmake:36 (message):
  ARCH not defined.  Check that BOARD=imx8mn_evk, is selecting an appropriate
  SoC in Kconfig, SoC=, and that the SoC is selecting the correct
  architecture.
Call Stack (most recent call first):
  /home/emilio/zephyrproject/zephyr/cmake/modules/zephyr_default.cmake:133 (include)
  /home/emilio/zephyrproject/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:66 (include)
  /home/emilio/zephyrproject/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:92 (include_boilerplate)
  CMakeLists.txt:5 (find_package)


-- Configuring incomplete, errors occurred!
FATAL ERROR: command exited with status 1: /usr/bin/cmake -DWEST_PYTHON=/home/emilio/zephyrproject/.venv/bin/python3 -B/home/emilio/zephyrproject/build -GNinja -DBOARD=imx8mn_evk/mimx8mn6/m7 -S/home/emilio/zephyrproject/zephyr/samples/hello_world

**Environment (please complete the following information):**
 - OS: Linux
 - Toolchain: Zephyr SDK
 - Commit SHA: 1fe904539d32b5b7d18361d63acc7a47c9171400

--- a/soc/nxp/imx/soc.yml
+++ b/soc/nxp/imx/soc.yml
@@ -6,23 +6,16 @@ family:
     - name: mimx8qm6
       cpuclusters:
       - name: adsp
-      - name: a72
-      - name: a53
-      - name: m4
   - name: imx8x
     socs:
     - name: mimx8qx6
       cpuclusters:
       - name: adsp
-      - name: a35
-      - name: m4
   - name: imx8ulp
     socs:
     - name: mimx8ud7
       cpuclusters:
       - name: adsp
-      - name: f1_dsp
-      - name: a35
   - name: imx8m
     socs:
     - name: mimx8ml8
@@ -37,10 +30,8 @@ family:
     - name: mimx8mn6
       cpuclusters:
       - name: a53
-      - name: m7
     - name: mimx8mq6
       cpuclusters:
-      - name: a53
       - name: m4
   - name: imx9
     socs:

